### ISSUE TITLE
TimeseriesGenerator introduces offset between target and test data

### ISSUE THREAD

**Issue by JimBiardCics** [keras-team/keras-preprocessing#66]
If you inspect the indices of the target and test data provided with each iteration of TimeseriesGenerator, you find that the target data comes from time step ```i```, while the test data comes from time steps ```i-length``` to ```i-1```, inclusive. There appears to be no way to adjust this offset.

The line

https://github.com/keras-team/keras-preprocessing/blob/2c7ef1d27bff4a24fe8bd2dc3800ccdebfc08a75/keras_preprocessing/sequence.py#L379

should be changed to

``` targets[j] = self.targets[rows[j]-self.sampling_rate] ```

or something similar.
Here's some sample code that displays the issue.

```
from __future__ import print_function

from keras.preprocessing.sequence import TimeseriesGenerator

import numpy


target = numpy.zeros((100,4,4), dtype = numpy.float32)

for i in range(0,100):
    target[i,...] = i

test = 0 + target

sequence = TimeseriesGenerator(test, target, length = 5, sampling_rate = 1,
                               stride = 1, start_index = 0, end_index = None,
                               shuffle = False, batch_size = 32)

epochs = len(sequence)

print('Length of sequence is', epochs)

epoch = 1

for block in sequence:
    print('Epoch', epoch)
    print('    test data')
    print('        shape    =', block[0].shape)
    print('        elements =', block[0][:,:,2,2])
    print('    target data')
    print('        shape    =', block[1].shape)
    print('        elements =', block[1][:,2,2])

    epoch += 1
```

**Comment by jberglinds** [comment]
I just noticed the same thing. 
An offset seems to be introduced between inputs and targets. 
I have a single feature input array, and a target that is the same array, shifted by 1 index. 
After running TimeseriesGenerator and inspecting the results, the target is the input shifted by 2 indices.

Minimal example:
```python
from keras.preprocessing.sequence import TimeseriesGenerator
input = [1, 2, 3, 4, 5]
print(TimeseriesGenerator(input, input, length=1)[0])
# Outputs: ([[1], [2], [3], [4]], [2, 3, 4, 5])
```
I would have expected the output to be `([[1], [2], [3], [4], [5]], [1, 2, 3, 4, 5])` in this case.

**Comment by noklam** [comment]
Just to add another example. The output is not correct when length%sampling_rate is not 0.
@jberglinds I actually don't know this generator can be used for multi-time step output?
```
from keras.preprocessing.sequence import TimeseriesGenerator

data = np.array([[i] for i in range(20)])

data_generator = TimeseriesGenerator(data, data,
                    
                                        length=10, sampling_rate=2, stride=7, batch_size=2, shuffle=False)
next(iter(data_generator))

data_generator = TimeseriesGenerator(data, data,
                    
                                        length=11, sampling_rate=2, stride=7, batch_size=2, shuffle=False)
next(iter(data_generator))
```
![image](https://user-images.githubusercontent.com/18221871/54583852-7dcdf700-4a50-11e9-814e-a1ca37f2d1ab.png)

**Comment by awilliamson** [comment]
Correct me if I'm wrong, but isn't this based on the assumption that the sequences generated have their data label `+1` from their location? I've had issues related to this, resulting in low validation as it's choosing the wrong metrics. Perhaps I need to reformat my input data.

**Comment by RobAltena** [comment]
All the unit tests pass, so this looks like the intended behavior. Label[i] maps to data[i-L : i], 

The current documentation for targets:
 _targets: Targets corresponding to timesteps in data. It should have same length as data._

I agree that this kind of implies that label[i] corresponds to the ending at data[i]. Can we fix this by updating the documentation? (I am happy to contribute the PR).
